### ISSUE TITLE
Assertion failed: (I != ValueMap.end() && "Value not in slotcalculator!")

### ISSUE THREAD

**Issue by will** [crystal-lang/crystal#2614]
```
Assertion failed: (I != ValueMap.end() && "Value not in slotcalculator!"), function getValueID, file /tmp/llvm20151214-69554-hagc7s/llvm-3.6.2.src/lib/Bitcode/Writer/ValueEnumerator.cpp, line 409.
```

Seeing this with crystal 6c8be57 and crystal-pg cafe5a1, but everything runs fine seems more like a warning.

**Comment by asterite** [comment]
What happens when you compile with `VERIFY=1`?

**Comment by asterite** [comment]
I can't seem to reproduce it. Did you compile 6c8be573397981bcdd355df7cfab12da84cbb10b with Crystal 0.16.0 or 0.17.0?

**Comment by asterite** [comment]
Finally, how to get that assertion failed? I did `ccrystal spec` at crystal-pg, both at HEAD and the revision you mention.

**Comment by asterite** [comment]
Oh, with VERIFY=1 I also get the same error... Definitely a bug. I should have tried with that flag...

**Comment by asterite** [comment]
\*Sigh\*... I guess I'll release 0.17.2 soon :-)

**Comment by will** [comment]
You did reproduce it then?

Compiled with 0.17.0

**Comment by asterite** [comment]
Yes. This reproduces it:

```crystal
class Foo
  BAR = 42

  @z : Int32

  def initialize
    @z = pointerof(BAR).value
  end

  def z
    @z
  end
end

Foo.new.z
```

Compiled with VERIFY=1 it gives an error.

Interestingly, this crashes 0.16.0: https://play.crystal-lang.org/#/r/yuc

**Comment by asterite** [comment]
@will I just pushed a fix to the release/0.17 branch. Could you check that it works well now?

I was able to compile fine but can't connect to the amazon instance anymore, but I don't know if it's shut down now.

I also compiled a few other projects with VERIFY=1 and they all compile fine. I'll probably release 0.17.2 tomorrow with this fix (I'll wait a bit in case you find other bugs 😄)

**Comment by will** [comment]
@asterite crystal/d8895a4 looks good, specs work with/without VERIFY=1 and the openssl is working.

**Comment by asterite** [comment]
@will \o/ 

I guess I'll start the release now then. I hope the homebrew team doesn't mind accepting another PR after such a small time.

Final question: what about 5e83d266b73b1e8912bb52f61fe1a71cd231ad60 ? It's the revision I'll tag as 0.17.2

**Comment by will** [comment]
5e83d26 is good

**Comment by asterite** [comment]
I'm closing this as it's fixed in 0.17.2
